gqcp/include/Basis/Transformations/UOrbitalRotationGenerators.hpp

```diff
@@ -0,0 +1,133 @@
+
+        // The total number of orbitals determines the size of the total kappa matrix. The alpha and beta component are created separately.
+        SquareMatrix<Scalar> kappa_alpha_full_matrix = SquareMatrix<Scalar>::Zero(K);
+        SquareMatrix<Scalar> kappa_beta_full_matrix = SquareMatrix<Scalar>::Zero(K);
+
+        // Depending on the row and column occupation types, we fill in the correct block of the total kappa matrix and leave the rest to be zero.
+        if (row_occupation_type == OccupationType::k_occupied && column_occupation_type == OccupationType::k_occupied) {
+            kappa_alpha_full_matrix.topLeftCorner(generators.alpha().numberOfOrbitals(), generators.alpha().numberOfOrbitals()) = generators.alpha().asMatrix();
+            kappa_beta_full_matrix.topLeftCorner(generators.beta().numberOfOrbitals(), generators.beta().numberOfOrbitals()) = generators.beta().asMatrix();
+        } else if (row_occupation_type == OccupationType::k_occupied && column_occupation_type == OccupationType::k_virtual) {
+            kappa_alpha_full_matrix.topRightCorner(generators.alpha().numberOfOrbitals(), generators.alpha().numberOfOrbitals()) = generators.alpha().asMatrix();
+            kappa_beta_full_matrix.topRightCorner(generators.beta().numberOfOrbitals(), generators.beta().numberOfOrbitals()) = generators.beta().asMatrix();
+        } else if (row_occupation_type == OccupationType::k_virtual && column_occupation_type == OccupationType::k_occupied) {
+            kappa_alpha_full_matrix.bottomLeftCorner(generators.alpha().numberOfOrbitals(), generators.alpha().numberOfOrbitals()) = generators.alpha().asMatrix();
+            kappa_beta_full_matrix.bottomLeftCorner(generators.beta().numberOfOrbitals(), generators.beta().numberOfOrbitals()) = generators.beta().asMatrix();
+        } else {
+            kappa_alpha_full_matrix.bottomRightCorner(generators.alpha().numberOfOrbitals(), generators.alpha().numberOfOrbitals()) = generators.alpha().asMatrix();
+            kappa_beta_full_matrix.bottomRightCorner(generators.beta().numberOfOrbitals(), generators.beta().numberOfOrbitals()) = generators.beta().asMatrix();
+        }
+
+        return UOrbitalRotationGenerators<Scalar> {UOrbitalRotationGeneratorsComponent<Scalar> {kappa_alpha_full_matrix}, UOrbitalRotationGeneratorsComponent<Scalar> {kappa_beta_full_matrix}};
+    }
+
+
+    /*
+     *  MARK: Access
+     */
+
+    /**
+     *  @return The anti-Hermitian orbital rotation generator spin resolved matrix kappa.
+     */
+    const SpinResolved<SquareMatrix<Scalar>> asMatrix() const {
+
+        return SpinResolved<SquareMatrix<Scalar>> {this->alpha().asMatrix(), this->beta().asMatrix()};
+    }
+
+    /**
+     *  @return The orbital rotation generators as the strict lower triangle of the spin resolved kappa matrix.
+     */
+    const SpinResolved<VectorX<Scalar>>& asVector() const {
```

Comment:
```suggestion
    SpinResolved<VectorX<Scalar>> asVector() const {
```

You can't use a reference type here, since you're creating a new object on the stack in this function.

Reply:
Oh right, as it is returning a spin resolved object and the parent class only has a vector as a member. My mistake, thanks for fixing it!